doomed_patient/doomed_patient/NPC.cpp

```diff
@@ -0,0 +1,101 @@
+#include "stdafx.h"
+#include "NPC.h"
+#include "PatientGame.h"
+#include "LevelCell.h"
+
+
+NPC::NPC(PatientGame* game, VectorXY startCoordinates, Texture * sprite)
+	: Character(game, startCoordinates, sprite)
+{
+	int random = rand() % 4;
+	movementDirection = static_cast<Directions::Direction>(random);
+}
+void NPC::move(Directions::Direction direction)
+{
+	centre += Directions::getDirectionVector(direction) * speed;
+}
+void NPC::update(LevelCell::Passable blocked)
+{
+	//Checking for passable directions and updateing direction
+
+	//North -> South
+
+	if (movementDirection == Directions::Direction::NORTH && !blocked.NORTH)
+	{ 
+		move(movementDirection);
+	}
+	else if (movementDirection == Directions::Direction::NORTH && blocked.NORTH && !blocked.SOUTH)
+	{
+		movementDirection = Directions::Direction::SOUTH;
```

Comment:
Since a lot of these ```if``` statements result in a reversed direction, you might want to check out using ```Directions::getOpposite(Direction direction)```